src/main/kotlin/domain/turn/IntermediateTurn.kt

```diff
@@ -0,0 +1,36 @@
+package domain.turn
+
+import domain.card.CardDeck
+import domain.gamer.Gamer
+import domain.gamer.Gamers
+import domain.gamer.player.Player
+
+object IntermediateTurn : Turn {
+
+    override fun proceed(
+        gamers: Gamers,
+        cardDeck: CardDeck,
+        changeState: (Turn) -> Unit,
+        askPlayerWantToStay: ((String) -> Boolean)?,
+    ) {
+        gamers.gamerToHit()?.let {
+            if (it is Player && askPlayerWantToStay?.invoke(it.name) == true) {
+                proceedStay(it, changeState)
+                return
+            }
+            proceedHit(it, cardDeck, changeState)
+        } ?: run {
+            changeState(FinalTurn)
+        }
```

Comment:
범위 지정 함수대신 엘비스 연산자를 사용해 null check를 하면 어떨까요?
```suggestion
        val hitGamer = gamers.gamerToHit() ?: run {
            changeState(FinalTurn)
            return
        }

        if (hitGamer is Player && askPlayerWantToStay?.invoke(hitGamer.name) == true) {
            proceedStay(hitGamer, changeState)
            return
        }
        proceedHit(hitGamer, cardDeck, changeState)
```